great_expectations/expectations/registry.py

```diff
@@ -95,48 +94,6 @@ def register_expectation(expectation: Type["Expectation"]) -> None:  # noqa: F82
     _registered_expectations[expectation_type] = expectation
 
 
-def register_data_assistant(
```

Comment:
Purpose of refactor - This are functions but they could be methods. They directly relate to activities around registration, which is encapsulated within the `DataAssistantDispatcher`. Wherever possible, let's try to associate functions with classes that cover that particular behavior (within reason of course).